scripts/clidocgen/main.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/spf13/cobra/doc"
+
+	"github.com/coder/coder/cli"
+)
+
+type route struct {
+	Title       string  `json:"title,omitempty"`
+	Description string  `json:"description,omitempty"`
+	Path        string  `json:"path,omitempty"`
+	IconPath    string  `json:"icon_path,omitempty"`
+	State       string  `json:"state,omitempty"`
+	Children    []route `json:"children,omitempty"`
+}
+
+type manifest struct {
+	Versions []string `json:"versions,omitempty"`
+	Routes   []route  `json:"routes,omitempty"`
+}
+
+func main() {
+	cmd := cli.Root(cli.AGPL())
+
+	// Get paths
+	basePath := os.Getenv("BASE_PATH")
+	if basePath == "" {
+		log.Fatal("BASE_PATH should be defined")
+	}
+	markdownDocsDir := path.Join(basePath, "docs/cli")
+	manifestFilepath := path.Join(basePath, "docs/manifest.json")
+
+	// Generate markdown
+	err := doc.GenMarkdownTree(cmd, markdownDocsDir)
+	if err != nil {
+		log.Fatal("Error on generating CLI markdown docs: ", err)
+	}
+
+	// Create CLI routes
+	cliRoutes := []route{}
```

Comment:
nit: var cliRoutes []route

Reply:
From someone who is not a golang developer, is there any diff? Or is just a different style we use in the codebase?

Reply:
In this case, it's more like the clean code, but there are [differences](https://stackoverflow.com/questions/29164375/correct-way-to-initialize-empty-slice). There is a good [write-up](https://go.dev/doc/effective_go) about writing good quality Go code, you may find it useful. BTW if there is a similar document for TypeScript, I'd love to read it too 👍

Reply:
Thank you! In the TS world, a lot of things are more "preferences" than practical differences to be honest so it depends a lot on what conventions you are going to use. Usually, those conventions are defined as ESLint rules and provided by big companies like Airbnb, Facebook, Microsoft, Google, etc. I personally like to use the default style options from ESLint and add the custom ones based on the team's taste or recommendation.

Reply:
Also, should we enforce the way you recommended initializing a slice in the `golint` rules?

Reply:
I remember that I have always seen this check enabled, maybe there is a strong reason that it's disabled for Coder?

Reply:
I have no idea 🤷‍♂️

Reply:
Let's leave it for now and focus on the main goal of this PR 👍